### PR TITLE
Additional UI components

### PR DESCRIPTION
**Pull request by nidhi-kala**
Addressing #496

### Adding the following components:
- [x] Accordion
- [x] Anchor Button (Button styled href)
- [x] Modal (with the props: open, title)

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

Accordion:
Screenshot 2023-04-28 at 00 04 32

Anchor Button:
Screenshot 2023-04-28 at 00 05 25

Modal:
Screenshot 2023-04-28 at 00 03 01

### CODE REVIEW

File: .changeset/violet-clouds-tell.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@evidence-dev/preprocess': minor
+'@evidence-dev/components': minor
+---
+
+Addtional UI components added
```
Comment (nidhi-kala) [inline .changeset/violet-clouds-tell.md:6]: Additional UI Components added: Accordion, Anchor Button , Modal

**Comment by hughess** [comment]
Hey @nidhi-kala, I checked into the error with the vercel deploy preview and it looks you've got imports in two places - once in preprocess and again in the markdown files. Should work once you remove the markdown imports!

**Comment by nidhi-kala** [comment]
> Hey @nidhi-kala, I checked into the error with the vercel deploy preview and it looks you've got imports in two places - once in preprocess and again in the markdown files. Should work once you remove the markdown imports!

Oh yeah! I was actually just working on this. I will update it. Thanks! :)

**Review by archiewood — COMMENTED** [review]
Looking great, really excited to add some of these to the project!

Know you are still working on them, so might change etc. Just an initial thought on syntax.

### Accordion
Typically we aim for an experience so that users can write in "pure" markdown + components - ie no script tags. Occasionally advanced users may "escape" into using them, but we try and avoid it being _required_ keep things simple!

The accordion component you have here I think is pretty hard to use without opening a script tag (you can pass the accordianItems in as a prop, but it will get pretty unwieldy)

I wonder if there is an approach similar to the [Tabs component](https://github.com/evidence-dev/evidence/tree/main/sites/example-project/src/components/ui/Tabs), eg

```
<Accordion>
    <AccordionItem title="Item 1">
          This is the first item's accordion body. It will remain hidden by default
    <AccordionItem/>
    <AccordionItem title="Item 2">
          This is the second item's accordion body
    <AccordionItem/>
<Accordion>
```

**Comment by nidhi-kala** [comment]
@archiewood Thank you for the early feedback and pointing me to the tabs component :)
Initially I wanted to make sure basic accordion was working and then iterate on it to have a nested component. I've update the PR now. Let me know if it needs any changes! 

I'll add footnotes component tomorrow :-D

File: sites/example-project/src/components/ui/AnchorButton.svelte [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<script>
+	export let href;
+</script>
+
+<a
+	{href}
+	class="font-ui font-normal text-base no-underline border mt-5 rounded-lg py-2 px-3 inline-block hover:border-blue-200 hover:bg-blue-100 hover:no-underline hover:text-gray-800 focus:text-gray-900 focus:border-blue-200"
```
Comment (archiewood) [inline sites/example-project/src/components/ui/AnchorButton.svelte:7]: Maybe a short transition for on hover style?

File: sites/example-project/src/components/ui/Modal.svelte [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	import CloseIcon from '$lib/icons/CloseIcon.svelte';
+	export let open = false;
+	export let title = '';
+	export let innerText = '';
+	const isOpen = () => {
+		open = !open;
+	};
+</script>
+
+<button on:click={() => isOpen()} class="border rounded px-4 py-1 text-sm my-3"> Open modal</button>
```
[the remark this answers is not in the document]
Comment (archiewood) [reply sites/example-project/src/components/ui/Modal.svelte:11]: Also, I think we should offer the user the ability to change the button text. `buttonText="Click Me"` ?

[the remark this answers is not in the document]
Comment (nidhi-kala) [reply sites/example-project/src/components/ui/Modal.svelte]: Sure, sorry I forgot to use the same styles as the AnchorButton, I've updated it now and added the buttonText prop as well.

Screenshot 2023-05-12 at 17 36 28

File: sites/example-project/src/components/ui/AccordionItem.svelte [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+<!-- svelte-ignore a11y-click-events-have-key-events -->
+<div bind:this={node} on:click={() => accordionContext.setActiveItem(index)} class="accordion-item">
+	<button
+		class="flex justify-between items-center w-full px-4 py-2 bg-white border-none cursor-pointer transition-colors duration-300 hover:bg-gray-100 focus:outline-none"
+	>
+		<h3 class="text-lg">{title}</h3>
+		<ChevronToggle {toggled} vertical="true" size="20" />
+	</button>
+	<div
+		class="text-base overflow-hidden transition-all duration-300 max-h-0 {toggled
```
[the remark this answers is not in the document]
Comment (nidhi-kala) [reply sites/example-project/src/components/ui/AccordionItem.svelte]: I think, transitions work best with absolute height, so using max-height(or auto-height) might be an issue. I've tried to make it smoother by measuring the height dynamically. 
If you notice any issues, let me know and I'll look into it.

File: sites/example-project/src/components/ui/AccordionItem.svelte [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		index = accordionItems.indexOf(node);
+	});
+</script>
+
+<!-- svelte-ignore a11y-click-events-have-key-events -->
+<div bind:this={node} on:click={() => accordionContext.setActiveItem(index)} class="accordion-item">
+	<button
+		class="flex justify-between items-center w-full px-4 py-2 bg-white border-none cursor-pointer transition-colors duration-300 hover:bg-gray-100 focus:outline-none"
+	>
+		<h3 class="text-lg">{title}</h3>
```
[the remark this answers is not in the document]
Comment (nidhi-kala) [reply sites/example-project/src/components/ui/AccordionItem.svelte:35]: fixed it, it had a padding issue.

File: sites/example-project/src/components/ui/AccordionItem.svelte [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		<ChevronToggle {toggled} vertical="true" size="20" />
+	</button>
+	<div
+		class="text-base overflow-hidden transition-all duration-300 max-h-0 {toggled
+			? 'max-h-80'
```
[the remark this answers is not in the document]
Comment (nidhi-kala) [reply sites/example-project/src/components/ui/AccordionItem.svelte]: Nope. I removed the `overflow-hidden` and `max-height`.  The content should not be clipped anymore.

File: sites/example-project/src/components/ui/AnchorButton.svelte (hunk 2 above)
Comment (nidhi-kala) [reply sites/example-project/src/components/ui/AnchorButton.svelte]: added that. 
Thanks for the review! :)

**Comment by archiewood** [comment]
Ugh somehome I didnt hit send on a few edits I suggested, sorry

Hoping I remember them all!

## Accordion
The rounding of the corners slightly cuts off the edges

## Anchor button
Looks great 

## Modal
Can we remove the border round the "X" button

## Docs
Let's add some. For Accordian and Modal, I think it would be helpful to add a couple of different examples!

File: packages/core-components/src/lib/unsorted/ui/Modal.svelte [hunk 7]
```diff
@@ -0,0 +1,62 @@
+<script context="module">
+	export const evidenceInclude = true;
+</script>
+
+<script>
+	import CloseIcon from '../icons/CloseIcon.svelte';
```
Comment (ItsMeBrianD) [inline packages/core-components/src/lib/unsorted/ui/Modal.svelte:6]: I mentioned this in another PR, and I don't know if we have a solid answer for this yet, but we want to normalize on an icon library:

https://github.com/evidence-dev/evidence/pull/955#discussion_r1232621678

File: packages/core-components/src/lib/unsorted/ui/Modal.svelte [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+	};
+</script>
+
+<button
+	on:click={() => isOpen()}
+	class="font-ui font-normal text-base no-underline border my-2 rounded-lg py-2 px-3 inline-block transition hover:ease-in hover:border-blue-200 hover:bg-blue-100 hover:no-underline hover:text-gray-800 focus:text-gray-900 focus:border-blue-200"
+	>{buttonText}</button
+>
+{#if open}
+	<!-- svelte-ignore a11y-click-events-have-key-events -->
```
Comment (ItsMeBrianD) [inline packages/core-components/src/lib/unsorted/ui/Modal.svelte:22]: I'm not super thrilled about having these ignored; for the accordion I think it's okay. Can we add a listener for the `escape` key to close the modal if it is open? That should resolve the issue and add a behavior that users will probably expect.

We also may want to include support for clickOutside, potentially as a prop to toggle it (default off), there is an action that can be used in the `core-components/src/clickOutside.js` file

File: packages/core-components/src/lib/unsorted/ui/Modal.svelte
Comment (ItsMeBrianD) [inline packages/core-components/src/lib/unsorted/ui/Modal.svelte:1]: image
Moving the padding up a level for the modal might make this look a little better, having no padding at the bottom of the modal can be a little jarring. Would love to hear @archiewood's thoughts on this as well

File: packages/core-components/src/lib/unsorted/ui/Modal.svelte (hunk 7 above)
Comment (ItsMeBrianD) [reply packages/core-components/src/lib/unsorted/ui/Modal.svelte:6]: I'm going to open a pull request today or tomorrow adding the icon library, once that is merged you can merge it into your branch and leverage that.

**Review by ItsMeBrianD — CHANGES_REQUESTED** [review]
This looks pretty good! 

There are a few visual things to tweak in the Modal, but the functionality is well done.

I would also love to see JSDoc added to the new components before they are added, so we don't need to go back and add them later. If you haven't use JSDoc before, there's some great documentation [on the typescript website](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html)

File: packages/core-components/src/lib/unsorted/ui/Modal.svelte
Comment (archiewood) [reply packages/core-components/src/lib/unsorted/ui/Modal.svelte:1]: Makes sense to me to match the padding

**Comment by hughess** [comment]
Took a spin through these and they're looking great!

A few things I noticed:

## Accordion
- I'm still seeing the corners cut off on the accordion component:
   CleanShot 2023-06-27 at 09 25 34@2x
- Is it possible to vertically centre the chevrons? They look a bit higher than the text at the moment:
   CleanShot 2023-06-27 at 09 25 50@2x
- The accordion closes when you click inside the area - could we have it remain open if the click is within the content area?
   - I think depending on the content inside the accordion, people may need to click in there

## Modal
- Same as accordion for clicking - clicking in the modal closes it, but users might need to click inside while keeping it open

## Anchor Button
- I would propose calling this one `Button` or `LinkButton` - rationale being that non-web devs won't know what an anchor is
- Under the same logic, I would propose changing the `href` prop to `url`

**Comment by archiewood** [comment]
Addressed all the above feedback except the clickOutside which for some reason eludes me

File: packages/core-components/src/lib/unsorted/ui/Modal.svelte (hunk 8 above)
Comment (archiewood) [reply packages/core-components/src/lib/unsorted/ui/Modal.svelte]: Svelte-ignore removed by making this a button rather than a div

**Comment by archiewood** [comment]
Thanks @nidhi-kala!

**Pull request merged**